README.md

```diff
@@ -1,23 +1,128 @@
+
+```
+pip install git+https://github.com/allen-cell-animated/colorizer-data.git@my-branch-name
+pip install git+https://github.com/allen-cell-animated/colorizer-data.git@v1.0.0
+```
+
+## Example Usage
+
+This is a simplified example. See the data scripts in [`timelapse-colorizer-data`](./timelapse-colorizer-data/) for complete working examples!
+
+```python
+import pandas as pd
+from colorizer_data.writer import (
+    ColorizerDatasetWriter,
+    ColorizerMetadata,
+    FeatureInfo,
+    FeatureType,
+)
+from colorizer_data.utils import (
+    configureLogging,
+    generate_frame_paths,
+    remap_segmented_image,
+    make_bounding_box_array,
+)
+
+# Open a dataset
+data: pd.DataFrame = pd.read_csv("dataset.csv")
+output_dir = "./data"
+
+configureLogging(output_dir)
+writer = ColorizerDatasetWriter(output_dir, data)
+
+# Write data and features from the dataset
+writer.write_data(
+    tracks=data["tracks"],
+    times=data["times"],
+    outliers=data["outliers"],
+    centroids_x=data["centroids x"],
+    centroids_y=data["centroids y"]
+)
+writer.write_feature(
+    data["feature1"].to_numpy(),
+    FeatureInfo(label="My Feature", type=FeatureType.CONTINUOUS)
+)
+writer.write_feature(
+    data["feature2"].to_numpy(),
+    FeatureInfo(label="My Other Feature", type=FeatureType.CATEGORICAL, categories=["A", "B", "C"])
+)
+
+# Write frames and bounding boxes
+grouped_data = data.groupby("time")
+bounds_arr = make_bounding_box_array(grouped_data)
+for group_name, frame in grouped_data:
+    row = frame.iloc[0]
+    frame_number = row["time"]
+    # Get segmentations as a 2D array
+    seg2d: np.ndarray = row["2d-segmentation-image"]
+
+    # Remap the 2D array and write to the dataset as an image
+    seg_remapped, lut = remap_segmented_image(
+        seg2d,
+        frame,
+        "object-ids",
+    )
+    writer.write_image(seg_remapped, frame_number)
+    update_bounding_box_data(bounds_arr, seg_remapped)
+
+writer.write_data(bounds=bound_arr)
+writer.set_frame_paths(generate_frame_paths(len(grouped_data)))
+
+# Write manifest and finish
+metadata = ColorizerMetadata(
+    frame_width=400,
+    frame_height=300,
+    frame_units="µm"
+    frame_duration_sec=5,
+)
+writer.write_manifest(metadata=metadata)
+```
+
+## Versioning
+
+Colorizer-data uses [semantic versioning](https://semver.org). All versions are tagged in this repository with `vX.Y.Z`, where `X`, `Y`, and `Z` correspond with major, minor, and patch version numbers. The API will be backwards-compatible within the same major version.
+
+Contributing developers can update the version number using the [`bump-my-version`](https://github.com/callowayproject/bump-my-version) Python tool, which will automatically tag commits. You can learn more about [git tagging here.](https://git-scm.com/book/en/v2/Git-Basics-Tagging)
+
+### Basic Usage:
 
 ```
-pip install .
+pip install --upgrade bump-my-version
+
+bump-my-version bump --tag -v --current-version [version] [type]
```

Comment:
please add the dry run option here. You should do a dry run first to make sure the changes are as expected.

Reply:
also, please give an explicit example so that people know what the proper form of [version], [type], and [new tag] should be.   Specifically, the tags need to begin with `v`.

Reply:
Because of the multi-file configuration options, you can have the latest version in this README file be updated also by the same script. (Referencing my comment above about how users should install from latest tagged version)

Reply:
> also, please give an explicit example so that people know what the proper form of [version], [type], and [new tag] should be. Specifically, the tags need to begin with `v`.

Yup! There's already an example down below, though I've updated it and the instructions.